nucleus/dataset.py

```diff
@@ -1414,11 +1429,76 @@ def items_and_annotation_generator(
             endpoint=f"dataset/{self.id}/exportForTrainingPage",
             result_key=EXPORT_FOR_TRAINING_KEY,
             page_size=10000,  # max ES page size
+            query=query,
         )
         for data in json_generator:
             for ia in convert_export_payload([data], has_predictions=False):
                 yield ia
 
+    def items_and_annotation_chip_generator(
+        self,
+        chip_size: int,
+        stride_size: int,
+        cache_directory: str,
```

Comment:
I think this should be optional

Reply:
If we are writing the images to a location, which I detail further below, it should be necessary

Reply:
But I mean the whole caching to file feature should be optional. I think this feature would be useful in some instances in which I wouldn't want it storing the chips anywhere, just generating them transiently.

Reply:
Synced live: going to specify local cache vs s3 cache